query_execution/QueryContext.hpp

```diff
@@ -472,6 +479,8 @@ class QueryContext {
   std::vector<std::unique_ptr<Tuple>> tuples_;
   std::vector<std::unordered_map<attribute_id, std::unique_ptr<const Scalar>>> update_groups_;
 
+  const std::size_t query_id_;
```

Comment:
This is my personal preference: put any `const` data members before any non-`const`.

But it is up to you.